homeassistant/components/tibber/services.py

```diff
@@ -0,0 +1,94 @@
+"""Services for Tibber integration."""
+
+from __future__ import annotations
+
+import datetime as dt
+from datetime import date, datetime
+from functools import partial
+from typing import Final
+
+import voluptuous as vol
+
+from homeassistant.core import (
+    HomeAssistant,
+    ServiceCall,
+    ServiceResponse,
+    SupportsResponse,
+    callback,
+)
+from homeassistant.exceptions import ServiceValidationError
+from homeassistant.util import dt as dt_util
+
+from .const import DOMAIN
+from .sensor import TibberDataCoordinator
+
+PRICE_SERVICE_NAME = "get_prices"
+ATTR_START: Final = "start"
+ATTR_END: Final = "end"
+
+SERVICE_SCHEMA: Final = vol.Schema(
+    {
+        vol.Optional(ATTR_START): str,
+        vol.Optional(ATTR_END): str,
+    }
+)
+
+
+async def __get_prices(call: ServiceCall, *, hass: HomeAssistant) -> ServiceResponse:
+    tibber_connection = hass.data[DOMAIN]
+    coordinator = TibberDataCoordinator(hass, tibber_connection)
+
+    for home in tibber_connection.get_homes(only_active=False):
+        tibber_home = home
```

Comment:
Yes, I was not able to attach the service to the dataupdatecoordinator linked to the config_entry like for example, Easyenergie does. 
Subscribing to the last home should only give a problem when there are multiple home in different day-ahead regions.

Reply:
I do not think that is a good solution.
Could you include the home id in the service input?

Reply:
For now I am not sure on how to do that. There are 2 options I guess:
1. use a text selector and let the user provide the correct string for the `home_id`
2. use a select/option selector and find a way to populate the options with the ids from `tibber_connection.get_home_ids()` -> but from what I know this is not possible since the options need to be specified in `services.yaml`

Reply:
Option 3 would be auto include all (active) homes in the same way as is common for weather forecasts. Perhaps add a check if the data differs between multiple homes, if not only show the first.